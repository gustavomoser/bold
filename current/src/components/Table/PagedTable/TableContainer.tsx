import React from 'react'

export const TableContainer = ({ children }) => {
  return <div>{children}</div>
}
