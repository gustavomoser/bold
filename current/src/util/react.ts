import { MutableRefObject, Ref } from 'react'

/**
 * Compose multiple react refs.
 *
 * @returns A new ref that passes down the Element to all ref arguments.
 */
export function composeRefs<T>(...refs: Array<Ref<T>>) {
  return (value: T) => refs.forEach(ref => setRef(ref, value))
}

/**
 * Changes the value of a ref.
 * The ref can be a function ref or a object ref.
 *
 * @param ref The ref to have its value changed.
 * @param value The new value.
 */
export function setRef<T>(ref: Ref<T>, value: any) {
  if (typeof ref === 'function') {
    ref(value)
  } else if (ref) {
    const mutableRef = ref as MutableRefObject<T>
    mutableRef.current = value
  }
}

/**
 * Compose multiple event handlers.
 *
 * @returns A new function that triggers all handlers passed as parameters, in order.
 */
export function composeHandlers(...handlers: Array<(...params: any) => void>) {
  return (...args: any[]) => handlers.forEach(handler => handler && handler(...args))
}

/**
 * Invokes the native `value` property setter of an element.
 * From https://github.com/facebook/react/issues/10135#issuecomment-401496776
 *
 * @param element The element to invoke the setter on.
 * @param value The value to be set.
 */
export function setNativeValue(element: HTMLElement, value: any) {
  const prototype = Object.getPrototypeOf(element)
  const { set: prototypeValueSetter = null } = Object.getOwnPropertyDescriptor(prototype, 'value') || {}
  const { set: valueSetter = null } = Object.getOwnPropertyDescriptor(element, 'value') || {}

  if (prototypeValueSetter && valueSetter !== prototypeValueSetter) {
    prototypeValueSetter.call(element, value)
  } else if (valueSetter) {
    valueSetter.call(element, value)
  } else {
    throw new Error('The given element does not have a value setter')
  }
}
