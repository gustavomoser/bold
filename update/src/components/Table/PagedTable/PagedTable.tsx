import React from 'react'

import { DataTable, DataTableProps } from '../DataTable/DataTable'
import { TableFooter } from '../TableFooter/TableFooter'

import { TableContainer } from './TableContainer'

export interface PagedTableProps<T> extends DataTableProps<T> {
  /**
   * Current page, 0-indexed.
   */
  page: number

  /**
   * Size of the current page.
   */
  size: number

  /**
   * Total number of pages.
   */
  totalPages: number

  /**
   * Total number of elements.
   */
  totalElements: number

  /**
   * Allows customization of table size options
   */
  sizeOptions?: number[]

  /**
   * Called when the current page is changed.
   * @param page The selected page.
   */
  onPageChange(page: number): void

  /**
   * Called when the current page size is changed.
   * @param size The selected page size.
   */
  onSizeChange(size: number): void
}

export function PagedTable<T>(props: PagedTableProps<T>) {
  const { onPageChange, onSizeChange, page, size, totalPages, totalElements, sizeOptions, ...rest } = props

  return (
    <TableContainer>
      <DataTable {...rest} />
      <TableFooter
        style={{ borderTop: 'none' }}
        page={page}
        pageSize={size}
        totalPages={totalPages}
        totalElements={totalElements}
        sizeOptions={sizeOptions}
        onPageChange={onPageChange}
        onSizeChange={onSizeChange}
      />
    </TableContainer>
  )
}
