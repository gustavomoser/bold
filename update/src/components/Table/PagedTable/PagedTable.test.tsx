import { render } from '@testing-library/react'
import React from 'react'

import { PagedTable } from './PagedTable'

interface Row {
  id: number
  name: string
  age: number
}

const rows: Row[] = [
  { id: 1, name: 'MARIA MACHADO DE JESUS', age: 42 },
  { id: 2, name: 'JOSÃ‰ DA SILVA MOREIRA', age: 34 },
  { id: 3, name: 'ALICE BARBOSA', age: 27 },
]

it('should render correctly', () => {
  const sortHandler = jest.fn()
  const pageHandler = jest.fn()
  const sizeHandler = jest.fn()

  const { container } = render(
    <PagedTable
      rows={rows}
      page={0}
      size={10}
      totalPages={1}
      totalElements={3}
      sort={['-id', 'name']}
      sizeOptions={[10, 30, 50, 100]}
      onSortChange={sortHandler}
      onPageChange={pageHandler}
      onSizeChange={sizeHandler}
      columns={[
        { name: 'id', header: 'ID', sortable: true, render: (row: Row) => row.id },
        { name: 'name', header: 'Name', sortable: true, render: (row: Row) => row.name },
        { name: 'age', header: 'Age', render: (row: Row) => row.age },
      ]}
    />
  )
  expect(container).toMatchSnapshot()
})

it('should render without sizeOptions', () => {
  const sortHandler = jest.fn()
  const pageHandler = jest.fn()
  const sizeHandler = jest.fn()

  const { container } = render(
    <PagedTable
      rows={rows}
      page={0}
      size={10}
      totalPages={1}
      totalElements={3}
      sort={['-id', 'name']}
      onSortChange={sortHandler}
      onPageChange={pageHandler}
      onSizeChange={sizeHandler}
      columns={[
        { name: 'id', header: 'ID', sortable: true, render: (row: Row) => row.id },
        { name: 'name', header: 'Name', sortable: true, render: (row: Row) => row.name },
        { name: 'age', header: 'Age', render: (row: Row) => row.age },
      ]}
    />
  )
  expect(container).toMatchSnapshot()
})

it('should render correctly when total elements is bigger than sizeOption', () => {
  const sortHandler = jest.fn()
  const pageHandler = jest.fn()
  const sizeHandler = jest.fn()

  const { container } = render(
    <PagedTable
      rows={rows}
      page={0}
      size={10}
      totalPages={2}
      totalElements={3}
      sort={['-id', 'name']}
      sizeOptions={[2]}
      onSortChange={sortHandler}
      onPageChange={pageHandler}
      onSizeChange={sizeHandler}
      columns={[
        { name: 'id', header: 'ID', sortable: true, render: (row: Row) => row.id },
        { name: 'name', header: 'Name', sortable: true, render: (row: Row) => row.name },
        { name: 'age', header: 'Age', render: (row: Row) => row.age },
      ]}
    />
  )
  expect(container).toMatchSnapshot()
})
