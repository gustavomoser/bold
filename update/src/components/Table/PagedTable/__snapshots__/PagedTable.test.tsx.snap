// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.emotion-0 {
  width: 100%;
  border-collapse: collapse;
  border: 1px solid #D3D4DD;
}

.emotion-0 tr:hover td {
  background: #F0F0F5;
}

.emotion-3 {
  text-align: left;
  padding: 0.5rem 1rem;
  border-bottom: 1px solid #D3D4DD;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.emotion-5 {
  margin-left: 0.25rem;
  font-size: 1rem;
  fill: #AAAAB9;
}

.emotion-7 {
  fill: #0069D0;
}

.emotion-16 {
  text-align: left;
  padding: 0 1rem;
  height: 2.5rem;
  border-bottom: 1px solid #D3D4DD;
}

.emotion-27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border: 1px solid #D3D4DD;
  height: 40px;
  border-top: none;
}

.emotion-28 {
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 1rem;
  padding-left: 1rem;
}

<div>
  <div>
    <table
      class="emotion-0"
    >
      <thead
        class="emotion-1"
      >
        <tr
          class="emotion-1"
        >
          <th
            class="emotion-3"
            data-name="id"
          >
            <span
              class="emotion-4"
            >
              ID
              <svg
                class="emotion-5"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
              >
                <polygon
                  class="emotion-1"
                  points="12 4.94 16.95 9.89 18.36 8.47 12 2.11 5.64 8.47 7.05 9.89 12 4.94"
                />
                <polygon
                  class="emotion-7"
                  points="12 19.06 7.05 14.11 5.64 15.53 12 21.89 18.36 15.53 16.95 14.11 12 19.06"
                />
              </svg>
            </span>
          </th>
          <th
            class="emotion-3"
            data-name="name"
          >
            <span
              class="emotion-4"
            >
              Name
              <svg
                class="emotion-5"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
              >
                <polygon
                  class="emotion-7"
                  points="12 4.94 16.95 9.89 18.36 8.47 12 2.11 5.64 8.47 7.05 9.89 12 4.94"
                />
                <polygon
                  class="emotion-1"
                  points="12 19.06 7.05 14.11 5.64 15.53 12 21.89 18.36 15.53 16.95 14.11 12 19.06"
                />
              </svg>
            </span>
          </th>
          <th
            class="emotion-3"
            data-name="age"
          >
            Age
          </th>
        </tr>
      </thead>
      <tbody
        class="emotion-1"
      >
        <tr
          class="emotion-1"
        >
          <td
            class="emotion-16"
          >
            1
          </td>
          <td
            class="emotion-16"
          >
            MARIA MACHADO DE JESUS
          </td>
          <td
            class="emotion-16"
          >
            42
          </td>
        </tr>
        <tr
          class="emotion-1"
        >
          <td
            class="emotion-16"
          >
            2
          </td>
          <td
            class="emotion-16"
          >
            JOSÃ‰ DA SILVA MOREIRA
          </td>
          <td
            class="emotion-16"
          >
            34
          </td>
        </tr>
        <tr
          class="emotion-1"
        >
          <td
            class="emotion-16"
          >
            3
          </td>
          <td
            class="emotion-16"
          >
            ALICE BARBOSA
          </td>
          <td
            class="emotion-16"
          >
            27
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="emotion-27"
    >
      <span
        class="emotion-28"
      >
        <span>
          3 results
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`should render correctly when total elements is bigger than sizeOption 1`] = `
.emotion-0 {
  width: 100%;
  border-collapse: collapse;
  border: 1px solid #D3D4DD;
}

.emotion-0 tr:hover td {
  background: #F0F0F5;
}

.emotion-3 {
  text-align: left;
  padding: 0.5rem 1rem;
  border-bottom: 1px solid #D3D4DD;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.emotion-5 {
  margin-left: 0.25rem;
  font-size: 1rem;
  fill: #AAAAB9;
}

.emotion-7 {
  fill: #0069D0;
}

.emotion-16 {
  text-align: left;
  padding: 0 1rem;
  height: 2.5rem;
  border-bottom: 1px solid #D3D4DD;
}

.emotion-27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border: 1px solid #D3D4DD;
  height: 40px;
  border-top: none;
}

.emotion-28 {
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 1rem;
  padding-left: 1rem;
}

.emotion-29 {
  border-left: 1px solid #D3D4DD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 4px 1rem;
}

.emotion-30 {
  display: grid;
  grid-auto-flow: column;
  grid-auto-columns: minmax(min-content, max-content);
  grid-gap: 0.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-31 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5;
  color: #24252E;
}

.emotion-32 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
  padding: 0 0.25rem;
}

.emotion-32>span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-32:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0, 0, 0, 0.1);
}

.emotion-32:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-32:not(:disabled):hover {
  background-color: rgba(58, 58, 71, 0.1);
}

.emotion-34 {
  fill: currentColor;
  font-size: 1.5rem;
  margin-left: 0.125rem;
}

.emotion-35 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
}

.emotion-36 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
  cursor: not-allowed;
  opacity: 0.5;
}

.emotion-36>span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-36:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0, 0, 0, 0.1);
}

.emotion-36:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-36:not(:disabled):hover {
  background-color: rgba(58, 58, 71, 0.1);
}

.emotion-37 {
  fill: currentColor;
  font-size: 1.5rem;
}

.emotion-38 {
  position: relative;
}

.emotion-39 {
  background-color: #FFFFFF;
  border: solid 1px #8F8FA2;
  border-radius: 2px;
  color: #24252E;
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1rem;
  margin: 0;
  padding: calc(0.5rem - 1px) 0.5rem;
  width: 100%;
  transition-property: box-shadow;
  transition-duration: .2s;
  width: 68px;
  text-align: center;
  margin: 0 0.5rem 0 0.25rem;
}

.emotion-39[type="search"]::-webkit-search-decoration {
  display: none;
}

.emotion-39[type="search"]::-webkit-search-cancel-button {
  display: none;
}

.emotion-39[type="search"]::-webkit-search-results-button {
  display: none;
}

.emotion-39[type="search"]::-webkit-search-results-decoration {
  display: none;
}

.emotion-39:required {
  box-shadow: none;
}

.emotion-39::-webkit-input-placeholder {
  color: #696979;
}

.emotion-39::-moz-placeholder {
  color: #696979;
}

.emotion-39:-ms-input-placeholder {
  color: #696979;
}

.emotion-39::placeholder {
  color: #696979;
}

.emotion-39:disabled {
  border-color: #D3D4DD;
  background-color: #F0F0F5;
}

.emotion-39:not(:disabled):hover {
  border-color: #7A7B8E;
}

.emotion-39:not(:disabled):focus {
  outline: none;
  box-shadow: 0 0 0 2px #F0F0F5,0 0 0 4px #0069D0;
}

.emotion-39:not(:disabled):active {
  border-color: #0069D0;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.09);
}

.emotion-41 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
  margin-left: 0.25rem;
}

.emotion-41>span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-41:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0, 0, 0, 0.1);
}

.emotion-41:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-41:not(:disabled):hover {
  background-color: rgba(58, 58, 71, 0.1);
}

<div>
  <div>
    <table
      class="emotion-0"
    >
      <thead
        class="emotion-1"
      >
        <tr
          class="emotion-1"
        >
          <th
            class="emotion-3"
            data-name="id"
          >
            <span
              class="emotion-4"
            >
              ID
              <svg
                class="emotion-5"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
              >
                <polygon
                  class="emotion-1"
                  points="12 4.94 16.95 9.89 18.36 8.47 12 2.11 5.64 8.47 7.05 9.89 12 4.94"
                />
                <polygon
                  class="emotion-7"
                  points="12 19.06 7.05 14.11 5.64 15.53 12 21.89 18.36 15.53 16.95 14.11 12 19.06"
                />
              </svg>
            </span>
          </th>
          <th
            class="emotion-3"
            data-name="name"
          >
            <span
              class="emotion-4"
            >
              Name
              <svg
                class="emotion-5"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
              >
                <polygon
                  class="emotion-7"
                  points="12 4.94 16.95 9.89 18.36 8.47 12 2.11 5.64 8.47 7.05 9.89 12 4.94"
                />
                <polygon
                  class="emotion-1"
                  points="12 19.06 7.05 14.11 5.64 15.53 12 21.89 18.36 15.53 16.95 14.11 12 19.06"
                />
              </svg>
            </span>
          </th>
          <th
            class="emotion-3"
            data-name="age"
          >
            Age
          </th>
        </tr>
      </thead>
      <tbody
        class="emotion-1"
      >
        <tr
          class="emotion-1"
        >
          <td
            class="emotion-16"
          >
            1
          </td>
          <td
            class="emotion-16"
          >
            MARIA MACHADO DE JESUS
          </td>
          <td
            class="emotion-16"
          >
            42
          </td>
        </tr>
        <tr
          class="emotion-1"
        >
          <td
            class="emotion-16"
          >
            2
          </td>
          <td
            class="emotion-16"
          >
            JOSÃ‰ DA SILVA MOREIRA
          </td>
          <td
            class="emotion-16"
          >
            34
          </td>
        </tr>
        <tr
          class="emotion-1"
        >
          <td
            class="emotion-16"
          >
            3
          </td>
          <td
            class="emotion-16"
          >
            ALICE BARBOSA
          </td>
          <td
            class="emotion-16"
          >
            27
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="emotion-27"
    >
      <span
        class="emotion-28"
      >
        <span>
          3 results
        </span>
      </span>
      <div
        class="emotion-29"
      >
        <div
          class="emotion-30"
        >
          <span
            class="emotion-31"
          >
            Mostrar:
          </span>
          <button
            aria-haspopup="true"
            class="emotion-32"
            type="button"
          >
            <span>
              <span
                class="emotion-31"
              >
                10
              </span>
              <svg
                aria-hidden="true"
                class="emotion-34"
                height="1em"
                role="img"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  clip-rule="evenodd"
                  d="M16.294 8.294a1 1 0 111.414 1.414l-5 5a1 1 0 01-1.414 0l-5.001-5a1 1 0 011.414-1.415l4.294 4.294 4.293-4.293z"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="emotion-35"
        >
          <button
            class="emotion-36"
            disabled=""
            title="Previous page"
            type="button"
          >
            <span>
              <svg
                aria-hidden="true"
                class="emotion-37"
                height="1em"
                role="img"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  clip-rule="evenodd"
                  d="M14.707 16.294a1 1 0 11-1.414 1.414l-5-5a1 1 0 010-1.414l5.001-5.001a1 1 0 011.414 1.414l-4.294 4.294 4.293 4.293z"
                />
              </svg>
            </span>
          </button>
          <div
            class="emotion-38"
          >
            <input
              class="emotion-39"
              maxlength="4"
              title="Current page"
              type="text"
              value="1"
            />
          </div>
          <span
            class="emotion-31"
          >
            of
             
            2
          </span>
          <button
            class="emotion-41"
            title="Next page"
            type="button"
          >
            <span>
              <svg
                aria-hidden="true"
                class="emotion-37"
                height="1em"
                role="img"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  clip-rule="evenodd"
                  d="M8.294 7.707a1 1 0 111.414-1.414l5 5a1 1 0 010 1.414l-5 5.001a1 1 0 01-1.415-1.414L12.587 12 8.294 7.707z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render without sizeOptions 1`] = `
.emotion-0 {
  width: 100%;
  border-collapse: collapse;
  border: 1px solid #D3D4DD;
}

.emotion-0 tr:hover td {
  background: #F0F0F5;
}

.emotion-3 {
  text-align: left;
  padding: 0.5rem 1rem;
  border-bottom: 1px solid #D3D4DD;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.emotion-5 {
  margin-left: 0.25rem;
  font-size: 1rem;
  fill: #AAAAB9;
}

.emotion-7 {
  fill: #0069D0;
}

.emotion-16 {
  text-align: left;
  padding: 0 1rem;
  height: 2.5rem;
  border-bottom: 1px solid #D3D4DD;
}

.emotion-27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border: 1px solid #D3D4DD;
  height: 40px;
  border-top: none;
}

.emotion-28 {
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 1rem;
  padding-left: 1rem;
}

<div>
  <div>
    <table
      class="emotion-0"
    >
      <thead
        class="emotion-1"
      >
        <tr
          class="emotion-1"
        >
          <th
            class="emotion-3"
            data-name="id"
          >
            <span
              class="emotion-4"
            >
              ID
              <svg
                class="emotion-5"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
              >
                <polygon
                  class="emotion-1"
                  points="12 4.94 16.95 9.89 18.36 8.47 12 2.11 5.64 8.47 7.05 9.89 12 4.94"
                />
                <polygon
                  class="emotion-7"
                  points="12 19.06 7.05 14.11 5.64 15.53 12 21.89 18.36 15.53 16.95 14.11 12 19.06"
                />
              </svg>
            </span>
          </th>
          <th
            class="emotion-3"
            data-name="name"
          >
            <span
              class="emotion-4"
            >
              Name
              <svg
                class="emotion-5"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
              >
                <polygon
                  class="emotion-7"
                  points="12 4.94 16.95 9.89 18.36 8.47 12 2.11 5.64 8.47 7.05 9.89 12 4.94"
                />
                <polygon
                  class="emotion-1"
                  points="12 19.06 7.05 14.11 5.64 15.53 12 21.89 18.36 15.53 16.95 14.11 12 19.06"
                />
              </svg>
            </span>
          </th>
          <th
            class="emotion-3"
            data-name="age"
          >
            Age
          </th>
        </tr>
      </thead>
      <tbody
        class="emotion-1"
      >
        <tr
          class="emotion-1"
        >
          <td
            class="emotion-16"
          >
            1
          </td>
          <td
            class="emotion-16"
          >
            MARIA MACHADO DE JESUS
          </td>
          <td
            class="emotion-16"
          >
            42
          </td>
        </tr>
        <tr
          class="emotion-1"
        >
          <td
            class="emotion-16"
          >
            2
          </td>
          <td
            class="emotion-16"
          >
            JOSÃ‰ DA SILVA MOREIRA
          </td>
          <td
            class="emotion-16"
          >
            34
          </td>
        </tr>
        <tr
          class="emotion-1"
        >
          <td
            class="emotion-16"
          >
            3
          </td>
          <td
            class="emotion-16"
          >
            ALICE BARBOSA
          </td>
          <td
            class="emotion-16"
          >
            27
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="emotion-27"
    >
      <span
        class="emotion-28"
      >
        <span>
          3 results
        </span>
      </span>
    </div>
  </div>
</div>
`;
